test/denormalized_has_many_test.rb

```diff
@@ -171,17 +171,6 @@ def test_returned_records_with_open_transactions_should_not_be_readonly
     end
   end
 
-  def test_respect_should_use_cache_from_embedded_records
```

Comment:
This test was added waaay in the back: https://github.com/Shopify/identity_cache/commit/74d91534b61bf254bbf5969760afe1a1fbdb3965

It asserts on behaviour that's opposite from what this PR does, so removing it.